app/models/app_feature_completion_migration.rb

```diff
@@ -0,0 +1,57 @@
+class AppFeatureCompletionMigration
+  include ActiveModel::Validations
+
+  attr_reader :app, :feature_gate, :version_before_type_cast
+
+  validate :feature_gate_must_exist
+
+  delegate :version, to: :app_feature_completion
+
+  def initialize(params)
+    @app = params[:app] || raise("Must provide app")
+    @feature_gate = params[:feature_gate]
+    app_feature_completion.version = @version_before_type_cast = params[:version]
+  end
+
+  def valid?
+    if destroy?
+      super
+    else
+      app_feature_completion.valid?
+    end
+  end
+
+  def errors
+    if destroy?
+      super
+    else
+      app_feature_completion.errors
+    end
+  end
+
+  def save
+    if destroy?
+      valid? && app_feature_completion.destroy.destroyed?
```

Comment:
is `destroy` not sufficiently truthy/falsey?

Reply:
This might be a mental tic, but destroy returns the model (which is why `destroyed?` is a thing). I'm not certain it would ever not return true from `destroyed?` but it seems like it'd be interesting if it didn't, and we should investigate, so I make it a 422.